src/opm/common/utility/TimeService.cpp

```diff
@@ -23,18 +23,33 @@
 #include <ctime>
 #include <utility>
 
-namespace {
-    std::time_t advance(const std::time_t tp, const double sec)
-    {
-        using namespace std::chrono;
+namespace Opm {
+namespace TimeService {
+
+std::time_t to_time_t(const time_point& tp) {
+    //return tp.time_since_epoch().count() / 1000;
+    return std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch()).count();
```

Comment:
This is undefined behaviour.  There's no guarantee that `.count()` of `time_since_epoch()` produces anything that's convertible to a `std::time_t`.  It happens to do the right thing in the current libstdc++ implementation but we can't depend on that.

Reply:
Well - it works currently - and the way I understand it it is guaranteed to work in C++20; I would say that is dependable.

Reply:
>> This is undefined behaviour.
>
> Well - it works currently

That's the nature of undefined behaviour.  And it doesn't actually "work".  It just happens to do what you expect on one particular implementation.

> the way I understand it it is guaranteed to work in C++20

Sure, but in C++20 we get **much** better date/time-handling tools prototyped in Howard Hinnant's date and timezone libraries and we won't need hacks like this.